packages/scene-composer/src/components/panels/scene-components/data-overlay/DataBindingMapEditor.spec.tsx

```diff
@@ -0,0 +1,160 @@
+    expect(addButton).not.toBeNull();
+
+    act(() => {
+      addButton!.click();
+    });
+
+    expect(onUpdateCallbackMock).toBeCalledTimes(1);
+    expect(onUpdateCallbackMock).toBeCalledWith({ valueDataBindings: [{ bindingName: 'random-uuid' }] });
+  });
+
+  it('should add additional binding by clicking add button without removing existing ones', async () => {
+    const { container } = render(
+      <DataBindingMapEditor
+        valueDataBindingProvider={mockProvider}
+        component={component}
+        onUpdateCallback={onUpdateCallbackMock}
+      />,
+    );
+    const polarisWrapper = wrapper(container);
+
+    const addButton = polarisWrapper.findButton('[data-test-id="add-binding-button"]');
+    expect(addButton).not.toBeNull();
+
+    act(() => {
+      addButton!.click();
+    });
+
+    expect(onUpdateCallbackMock).toBeCalledTimes(1);
+    expect(onUpdateCallbackMock).toBeCalledWith({
+      valueDataBindings: [...component.valueDataBindings, { bindingName: 'random-uuid' }],
+    });
+  });
+
+  it('should remove binding by clicking remove button', async () => {
+    const { container } = render(
+      <DataBindingMapEditor
+        valueDataBindingProvider={mockProvider}
+        component={component}
+        onUpdateCallback={onUpdateCallbackMock}
+      />,
+    );
+    const polarisWrapper = wrapper(container);
+
+    const removeButton = polarisWrapper.findButton('[data-test-id="remove-binding-button"]');
+    expect(removeButton).not.toBeNull();
+
+    act(() => {
+      removeButton!.click();
+    });
+
+    expect(onUpdateCallbackMock).toBeCalledTimes(1);
+    expect(onUpdateCallbackMock).toBeCalledWith({ ...component, valueDataBindings: [] }, true);
+  });
+
+  it('should call update when data binding changed', async () => {
+    render(
+      <DataBindingMapEditor
+        valueDataBindingProvider={mockProvider}
+        component={{ ...component, valueDataBindings: [...component.valueDataBindings, { bindingName: 'binding-2' }] }}
+        onUpdateCallback={onUpdateCallbackMock}
+      />,
+    );
+
+    builderOnChangeCb({ key: 'value' }, 1);
+
+    expect(onUpdateCallbackMock).toBeCalledTimes(1);
+    expect(onUpdateCallbackMock).toBeCalledWith(
+      {
+        ...component,
+        valueDataBindings: [
+          ...component.valueDataBindings,
+          { bindingName: 'binding-2', valueDataBinding: { key: 'value' } },
+        ],
+      },
+      true,
+    );
+  });
+
+  it('should update binding name', async () => {
+    const { container } = render(
+      <DataBindingMapEditor
+        valueDataBindingProvider={mockProvider}
+        component={component}
+        onUpdateCallback={onUpdateCallbackMock}
+      />,
+    );
+    const polarisWrapper = wrapper(container);
+
+    const nameInput = polarisWrapper.findInput('[data-test-id="binding-name-input"]');
+    expect(nameInput).not.toBeNull();
+
+    act(() => {
+      nameInput!.setInputValue('new-name');
+    });
```

Comment:
Separate your responsibilities instead of writing unit tests like integration tests.

You should:
- Verify DOM Structure using a snapshot test. You may want to test a few different property settings depending on what you need
- Verify event handlers/callbacks/effects through mocking, and direct execution.